src/main/java/com/impactupgrade/nucleus/util/Utils.java

```diff
@@ -209,14 +209,42 @@ public static List<Map<String, String>> getCsvData(String csv) throws IOExceptio
   }
 
   public static List<Map<String, String>> getCsvData(InputStream inputStream) throws IOException {
-    CsvMapper mapper = new CsvMapper();
-    CsvSchema schema = CsvSchema.emptySchema().withHeader();
-    MappingIterator<Map<String, String>> iterator = mapper.readerFor(Map.class).with(schema).readValues(inputStream);
-    List<Map<String, String>> result = new LinkedList<>();
-    while (iterator.hasNext()) {
-      result.add(iterator.next());
+    try (CSVParser csvParser = CSVParser.parse(
+        inputStream,
+        Charset.defaultCharset(),
+        CSVFormat.DEFAULT
+            .withIgnoreHeaderCase()
+            .withTrim()
+    )) {
+      CSVRecord headerRecord = csvParser.iterator().next();
+      List<String> headers = new ArrayList<>();
+      List<Integer> headerIndices = new ArrayList<>();
+
+      for (int i = 0; i < headerRecord.size(); i++) {
+        String header = headerRecord.get(i);
+        if (header != null && !header.trim().isEmpty()) {
+          headers.add(header.trim());
+          headerIndices.add(i);
+        }
+      }
```

Comment:
Instead of using the library to automatically parse the header and all rows, we're explicitly parsing the first row and using that as the valid header values. Then, below, we build up the Map iteratively, using that header and the column indices within it.